### ISSUE TITLE
unknown error,lose order

### ISSUE THREAD

**Issue by jeffie** [chirag04/react-native-in-app-utils#139]
in production env, i got an error callback, but the app store is already charged, so i cannot get an exact callback info ：
`{"code":"ESKERRORDOMAIN0","message":"无法连接到 iTunes Store","nativeStackIOS":["0   LanhaiApp                           0x000000010407d674 LanhaiApp + 300660","1   LanhaiApp                           0x000000010407d5d4 LanhaiApp + 300500","2   LanhaiApp                           0x00000001041a0d94 RCTFBQuickPerformanceLoggerConfigureHooks + 1178188","3   StoreKit                            0x00000001964d1278 <redacted> + 104","4   CoreFoundation                      0x0000000183091c20 CFArrayApplyFunction + 80","5   StoreKit                            0x00000001964d11f4 <redacted> + 148","6   StoreKit                            0x00000001964d222c <redacted> + 1220","7   StoreKit                            0x00000001964d2b3c <redacted> + 160","8   StoreKit                            0x00000001964d2460 <redacted> + 32","9   libdispatch.dylib                   0x0000000182b51088 <redacted> + 24","10  libdispatch.dylib                   0x0000000182b51048 <redacted> + 16","11  libdispatch.dylib                   0x0000000182b5db74 <redacted> + 1016","12  CoreFoundation                      0x0000000183173f20 <redacted> + 12","13  CoreFoundation                      0x0000000183171afc <redacted> + 2012","14  CoreFoundation                      0x00000001830922d8 CFRunLoopRunSpecific + 436","15  GraphicsServices                    0x0000000184f23f84 GSEventRunModal + 100","16  UIKit                               0x000000018c63f880 UIApplicationMain + 208","17  LanhaiApp                           0x0000000104038ec8 LanhaiApp + 20168","18  libdyld.dylib                       0x0000000182bb656c <redacted> + 4"],"domain":"SKErrorDomain","userInfo":{"NSLocalizedDescription":"无法连接到 iTunes Store"}}`

**Comment by lwansbrough** [comment]
@jeffie did you figure out what was causing this? I'm getting this error very often in production as well.

**Comment by chirag04** [comment]
From the description it might be possible that itunes store was down at the time. Let me know if you are seeing this consistently or are able to reproduce.

**Comment by lwansbrough** [comment]
I get this error often than I get purchase confirmations. This is from hundreds of buy presses/day.

**Comment by jeffie** [comment]
@chirag04 I cannot find the regular about it. In production,2-3 times monthly

**Comment by chirag04** [comment]
`2-3 times monthly` sounds like a legitimate app store failure. There is no way for me to confirm/debug this tho. will have to rely on you all debug this.

**Comment by jeffie** [comment]
@chirag04 Yes,I think it`s a app store failure, maybe there has an interrupt when purchase.But is there a right way to handle the result?because the callback result is incorrect
